tests/drivers/random/randomnumgen_basic_api/src/main.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <random.h>
+#include <ztest.h>
+
+/*
+ * @addtogroup t_randomnumgen_basic_api
+ * @{
+ * @defgroup t_random_get_entropy test_random_get_entropy
+ * @brief TestPurpose: verify Random number generator works
+ * @details
+ * - Test Steps
+ *   -# Read random numbers from RNG driver.
+ *   -# Verify whether buffer overflow occured or not.
+ *   -# Verify whether buffer completly filled or not.
+ * - Expected Results
+ *   -# Random number should be generated.
+ * @}
+ */
+
+#define BUFFER_LENGTH 10
+#define RECHECK_RANDOM_ENTROPY 0x10
+
+static int random_entropy(struct device *dev, char *buffer, char num)
+{
+	int ret, i;
+	int count = 0;
+
+	memset((char *)buffer, (char)num, BUFFER_LENGTH);
+
+	/* The BUFFER_LENGTH-1 is used so the driver will not
+	 * write the last byte of the buffer. If that last
+	 * byte is not 0 on return it means the driver wrote
+	 * outside the passed buffer, and that should never
+	 * happen.
+	 */
+	ret = random_get_entropy(dev, buffer, BUFFER_LENGTH-1);
+	if (ret) {
+		TC_PRINT("Error: random_get_entropy failed: %d\n", ret);
+		return TC_FAIL;
+	}
+	if (buffer[BUFFER_LENGTH-1] != num) {
+		TC_PRINT("Error: random_get_entropy buffer overflow\n");
+		return TC_FAIL;
+	}
+
+	for (i = 0; i < BUFFER_LENGTH-1; i++) {
+		TC_PRINT("  0x%02x\n", buffer[i]);
+		if (buffer[i] == num)
+			count++;
+	}
+
+	if (count >= 2)
+		return RECHECK_RANDOM_ENTROPY;
+	else
+		return TC_PASS;
+}
+
+/*
+ * Function invokes the get_entropy callback in driver
+ * to get the random data and fill to passed buffer
+ */
+static int get_entropy(void)
+{
+	struct device *dev;
+	u8_t buffer[BUFFER_LENGTH] = {0};
+	int ret;
+
+	TC_PRINT("Random Example! %s\n", CONFIG_ARCH);
```

Comment:
This is a test, not a sample program. Printing `CONFIG_ARCH` isn't necessary if we're printing the boot banner.

Reply:
The boot banner doesn't print the target architecture, just the OS version and build date.
I would happily approve a change to the boot banner to include this information however.